modules/core/src/main/scala/org/tesselation/infrastructure/db/Migrations.scala

```diff
@@ -0,0 +1,23 @@
+package org.tesselation.infrastructure.db
+
+import cats.effect.kernel.Sync
+import org.flywaydb.core.Flyway
+import org.flywaydb.core.api.output.MigrateResult
+import org.tesselation.infrastructure.db.doobie.DoobieDataSource
+
+trait Migrations[F[_]] {
+  def migrate: F[MigrateResult]
+}
+
+object Migrations {
+
+  def make[F[_]: Sync: DoobieDataSource]: Migrations[F] = new Migrations[F] {
+    private val flyway = Flyway
+      .configure()
+      .dataSource(DoobieDataSource[F].ds)
+      .load()
+
+    override def migrate: F[MigrateResult] = Sync[F].delay(flyway.migrate())
```

Comment:
is Sync really needed here? Async is the default class for effects in cats effects 3, not Sync.